account_move_date_period_sync/models/account_move.py

```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 PRAXYA (<http://praxya.com/>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, api
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    @api.onchange('date')
+    def _onchange_move_date(self):
+        if self.date:
+            period_obj = self.env['account.period']
+            period_id = period_obj.find(self.date)
```

Comment:
If the period is not in the system the funcion period_obj.find(self.date) return a warning like this. I think with this is enough.
![period-error](https://user-images.githubusercontent.com/19977041/27077285-d499336e-502f-11e7-91af-96d6a3bd369d.png)